src/mlpack/methods/ann/augmented/ham_unit_impl.hpp

```diff
@@ -123,7 +124,34 @@ void HAMUnit<E, J, S, W, C>::ResetParameters() {
   embed.ResetParameters();
   controller.ResetParameters();
   memory.ResetParameters();
-  RebuildParameters();
+
+  arma::mat embedParams = embed.Parameters();
+  arma::mat searchParams = search.Parameters();
+  arma::mat controllerParams = controller.Parameters();
+  arma::mat joinParams = memory.JoinObject().Parameters();
+  arma::mat writeParams = memory.WriteObject().Parameters();
+
+  size_t embedCount = embedParams.n_elem,
+         searchCount = searchParams.n_elem,
+         controllerCount = controllerParams.n_elem,
+         joinCount = joinParams.n_elem,
+         writeCount = writeParams.n_elem;
+
+  parameters = arma::mat(embedCount + searchCount + controllerCount + joinCount + writeCount, 1);
+
+  embed.Parameters() = arma::mat(parameters.memptr(), embedCount, 1, false, true);
```

Comment:
> Implemented Parameters() method as rcurtin described (make individual function parameters as memory pointers to some contiguous memory block), but there are still only zeros in the HAMUnit paramters. Can you take a look at the issue?

It should work if you use ``embed.Parameters() = arma::mat(parameters.memptr(), embedCount, 1, false, false);`` instead of ``embed.Parameters() = arma::mat(parameters.memptr(), embedCount, 1, false, true);``. Let us know if this works for you.